### ISSUE TITLE
x-errorbars don't appear on date axis

### ISSUE THREAD

**Issue by jonmmease** [plotly/plotly.js#3722]
Originally reported in https://github.com/plotly/plotly.py/issues/1488

Here's an example that specifies `error_x` and flips between setting the `xaxis.type` property between `'date'` and `'linear'`.

CodePen: https://codepen.io/jonmmease/pen/zXrmOK

```
var fig = {
  "data": [
    {
      "error_x": {
        "array": [
          100000.0,
          100000.0,
          100000.0
        ],
        "type": "data",
        "visible": true
      },
      "type": "scatter",
      "x": [
        1554370371547.085,
        1554370471547.085,
        1554370571547.085
      ],
      "y": [
        6,
        10,
        2
      ]
    }
  ],
  "layout": {
    "xaxis": {
      "type": "linear"
      // "type": "date"
    }
  }
}

Plotly.newPlot('myDiv', fig, {showSendToCloud: true});
```
with `xaxis.type: 'linear'`:
![newplot-7](https://user-images.githubusercontent.com/15064365/55545675-b1895d80-569b-11e9-905c-2280c551ffab.png)

with `xaxis.type: 'date'`:
![newplot-6](https://user-images.githubusercontent.com/15064365/55545696-b9490200-569b-11e9-9dae-10361c486a3d.png)

**Comment by etpinard** [comment]
Looks like we never implemented error bars on date axes:

https://github.com/plotly/plotly.js/blob/07822743b03e9bc1a3decd429c02b0452af49732/src/components/errorbars/calc.js#L37

**Comment by ArsumPB** [comment]
still nothing? need it very much for lightcurves :(

**Comment by ibrewster** [comment]
Any updates on this? Workarounds? I just got a graph I am supposed to make that uses error bars on dates. I guess I can simply draw lines as a separate plot (if there is a way to keep it out of the legend?), but that seems kludgy...

**Comment by tomaszps** [comment]
@ibrewster I think I eventually got around this using a gantt chart style plot, FYI. I think it's a kinda odd use case.

**Comment by James-Firth** [comment]
Chiming in that this would also be useful when charting user-recorded data points, we have a lot of uncertainty about when exactly events happened and want to show error bars

**Comment by gvwilson** [comment]
yes, but I expect it will be some time before it gets to the head of our queue :-/

**Comment by mshvarts** [comment]
Dang I just spend hours trying to figure out why it wasn't working by trying to debug the minified source code just to find this thread that it has never been implemented with dates 💀.  At least I didn't spend even longer than that .
But now what.. Do I need to implement this myself?

**Comment by tomaszps** [comment]
I would suggest to everyone who looks at this thread in the future to just
use the gant chart functions.

-Tomasz
